### PR TITLE
PIM-4354: Add a twig extension to know if attribute is granted with its code

### PR DESCRIPTION
**Pull request by willy-ahva**
| Q                 | A
| ----------------- | ---
| Specs             | Y
| Behats            | Y
| Blue CI           | Y
| Changelog updated | N/A
| Review and 2 GTM  | Y
| Micro Demo (PO)   | N/A
| Migration script  | N/A
| Tech Doc          | N/A

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/config/twig.yml [hunk 1]
```diff
@@ -48,6 +49,14 @@ services:
         tags:
             - { name: twig.extension }
 
+    pim_enrich.twig.attribute_granted:
```
Comment (filipsalpe) [inline src/Pim/Bundle/EnrichBundle/Resources/config/twig.yml]: all the other extension services end with "_extension", why not pim_enrich.twig.attribute_granted_extension?

Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/config/twig.yml]: hey @filipsalpe is in the place ! ;)

**Comment by willy-ahva** [comment]
GTM

File: src/Pim/Bundle/EnrichBundle/Twig/AttributeGrantedExtension.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+class AttributeGrantedExtension extends \Twig_Extension
+{
+    /** @var AuthorizationCheckerInterface */
+    protected $authorizationChecker;
+
+    /** @var AttributeRepositoryInterface */
+    protected $attributeRepository;
+
+    /**
+     * @param AuthorizationCheckerInterface $securityExtension
```
Comment (solivier) [inline src/Pim/Bundle/EnrichBundle/Twig/AttributeGrantedExtension.php]: doc is not the same securityExtension should be authorizationChecker

**Comment by solivier** [comment]
small comment then GTM for me

File: features/Context/WebUser.php [hunk 3]
```diff
@@ -1566,14 +1567,20 @@ public function iShouldNotSeeTheButton($button)
...
-            $buttonElement = $this
-            ->getCurrentPage()
-            ->find('css', sprintf('.modal a:contains("%s")', $buttonLabel));
-        }
+        $buttonElement = $this->spin(function () use ($buttonLabel) {
+            $node = $this
+                ->getCurrentPage()
+                ->find('css', sprintf('.ui-dialog button:contains("%s")', $buttonLabel));
+
+            if (!$node) {
```
Comment (phaseinducer) [inline features/Context/WebUser.php]: `null === $node`

File: features/Context/WebUser.php [hunk 4]
```diff
@@ -1566,14 +1567,20 @@ public function iShouldNotSeeTheButton($button)
...
+                $node = $this
+                    ->getCurrentPage()
+                    ->find('css', sprintf('.modal a:contains("%s")', $buttonLabel));
+            }
+
+            return $node;
+        });
```
Comment (phaseinducer) [inline features/Context/WebUser.php]: Specify a message please

File: src/Pim/Bundle/EnrichBundle/Twig/AttributeGrantedExtension.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Pim\Bundle\EnrichBundle\Twig;
+
+use Pim\Bundle\CatalogBundle\Repository\AttributeRepositoryInterface;
+use Symfony\Component\Security\Acl\Voter\FieldVote;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+/**
+ * Twig extension to know if edit an attribute is granted to a user
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Twig/AttributeGrantedExtension.php]: just edit ?

File: src/Pim/Bundle/EnrichBundle/Twig/AttributeGrantedExtension.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    public function isAttributeGranted($role, $attributeCode, $field = null)
+    {
+        $attribute = $this->attributeRepository->findOneByIdentifier($attributeCode);
+        if (null === $attribute) {
+            return false;
+        }
+        $attributeGroup = $attribute->getGroup();
+
+        if (null !== $field) {
+            $attributeGroup = new FieldVote($attributeGroup, $field);
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Twig/AttributeGrantedExtension.php]: A bit strange to reuse `$attribtueGroup` here

File: src/Pim/Bundle/EnrichBundle/Twig/AttributeGrantedExtension.php [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                'is_attribute_granted',
+                [$this, 'isAttributeGranted']
+            ),
+        ];
+    }
+
+    /**
+     * @param string $role
+     * @param string $attributeCode
+     * @param string $field
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Twig/AttributeGrantedExtension.php]: string|null

**Pull request merged**